test/common_utils.py

```diff
@@ -1054,6 +1054,23 @@ def random_fullrank_matrix_distinct_singular_value(l, *batches, **kwargs):
         return torch.stack(all_matrices).reshape(*(batches + (l, l)))
 
 
+def random_linalg_solve_processed_inputs(A_dims, b_dims, gen_fn, transform_fn, cast_fn):
```

Comment:
Once this PR is merged, I will use this function in other places in the test suite for `*solve` methods. This would reduce duplication of code.